itest/litd_custom_channels_test.go

```diff
@@ -2046,6 +2050,166 @@ func testCustomChannelsLiquidityEdgeCases(_ context.Context,
+	}()
+
+	select {
+	case <-done:
+		// If the payment completes before the slosh payment occurs then
+		// something went wrong, this is not the expected test case
+		// behavior.
+		t.Fatalf("payment completed before expected block period")
+	case <-timeoutChan:
+		// The expected time delay for the payment to complete has been
+		// passed, so now we wait for the payment to complete.
+		<-done
+	}
+
+	logBalance(t.t, nodes, assetID, "after htlc track cancel")
+
+	// Edge case: Charlie negotiates a quote with Dave which has a low max
+	// amount (~170k sats). Then Charlie creates an invoice with a total
+	// amount slightly larger than the max allowed in the quote (200k sats).
+	// Erin will try to pay that invoice with sats, in shards of max size
+	// 80k sats. Dave will eventually stop forwarding HTLCs as the RFQ HTLC
+	// tracking mechanism should stop them from being forwarded, as they
+	// violate the maximum allowed amount of the quote.
+
+	// Charlie starts by negotiating the quote.
+	res, err := charlieTap.RfqClient.AddAssetBuyOrder(
+		ctxb, &rfqrpc.AddAssetBuyOrderRequest{
+			AssetSpecifier: &rfqrpc.AssetSpecifier{
+				Id: &rfqrpc.AssetSpecifier_AssetId{
+					AssetId: assetID,
+				},
+			},
+			AssetMaxAmt:    10_000,
+			Expiry:         uint64(time.Now().Add(time.Hour).Unix()),
+			PeerPubKey:     dave.PubKey[:],
+			TimeoutSeconds: 10,
+		},
+	)
+	require.NoError(t.t, err)
+
+	quote, ok := res.Response.(*rfqrpc.AddAssetBuyOrderResponse_AcceptedQuote)
+	require.True(t.t, ok)
+
+	// We now manually add the invoice in order to inject the above,
+	// manually generated, quote.
+	iResp, err := charlie.AddInvoice(ctxb, &lnrpc.Invoice{
+		Memo:       "",
+		Value:      200_000,
+		RPreimage:  bytes.Repeat([]byte{11}, 32),
+		CltvExpiry: 60,
+		RouteHints: []*lnrpc.RouteHint{
+			&lnrpc.RouteHint{
+				HopHints: []*lnrpc.HopHint{
+					&lnrpc.HopHint{
+						NodeId: dave.PubKeyStr,
+						ChanId: quote.AcceptedQuote.Scid,
+					},
+				},
+			},
+		},
+	})
+	require.NoError(t.t, err)
+
+	// Now Erin tries to pay the invoice. Since the multipart payment will
+	// have some of its shards failing the pathfinding logic will keep going
+	// and we won't see a payment failure but a timeout. If a final outcome
+	// is not produced within a reasonable amount of time, we assume the
+	// payment is still trying to find a route, therefore the HTLC rejection
+	// works.
+	timeoutChan = time.After(PaymentTimeout / 2)
+	done = make(chan bool, 1)
+
+	ctxc, cancel := context.WithCancel(context.Background())
+
+	//nolint:lll
+	go func() {
+		// payInvoiceWithSatoshi(t.t, erin, iResp, lnrpc.Payment_FAILED)
+		sendReq := &routerrpc.SendPaymentRequest{
+			PaymentRequest:   iResp.PaymentRequest,
+			TimeoutSeconds:   int32(PaymentTimeout.Seconds()),
+			MaxShardSizeMsat: 80_000_000,
+			FeeLimitMsat:     1_000_000,
+		}
+		stream, err := erin.RouterClient.SendPaymentV2(ctxc, sendReq)
+		if err == nil {
+			_, _ = getPaymentResult(stream)
+		}
```

Comment:
I think you should use this patter here:
```
stream, err := erin.RouterClient.SendPaymentV2(ctxc, sendReq)
require.NoError(t.t, err)

_, _ = getPaymentResult(stream)
```

Need error check on `getPaymentResult` result?

Reply:
latest approach uses existing helpers, closing